### ISSUE TITLE
Fusil Fuzz: ProgressRecorderSession.pre_response is printing to the wrong stream

### ISSUE THREAD

**Issue by chfoo** [ArchiveTeam/wpull#15]
stdout:

```
INFO Fetching \u2018http://127.0.0.1:8898/robots.txt\u2019.
Requesting http://127.0.0.1:8898/robots.txt... ERROR Fatal exception.
Traceback (most recent call last):
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/engine.py", line 123, in _process_input
    yield self._process_url_item(url_item)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 505, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/engine.py", line 147, in _process_url_item
    session, url_item)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 505, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/engine.py", line 169, in _process_session
    response_factory=session.response_factory())
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 507, in run
    yielded = self.gen.send(next)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/http.py", line 638, in fetch
    raise response from response
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/http.py", line 524, in _process_request
    response = yield connection.fetch(request, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 505, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/http.py", line 278, in fetch
    response_factory)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 505, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/http.py", line 315, in _process_request
    response = yield self._read_response_header(response_factory)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 496, in run
    next = self.yield_point.get_result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 395, in get_result
    return self.runner.pop_result(self.key).result()
  File "/usr/local/lib/python3.3/dist-packages/tornado/concurrent.py", line 129, in result
    raise_exc_info(self.__exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.3/dist-packages/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 550, in inner
    self.set_result(key, result)
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 476, in set_result
    self.run()
  File "/usr/local/lib/python3.3/dist-packages/tornado/gen.py", line 507, in run
    yielded = self.gen.send(next)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/http.py", line 353, in _read_response_header
    self._events.pre_response.fire(response)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/actor.py", line 24, in fire
    handler(*args, **kargs)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/recorder.py", line 95, in pre_response
    session.pre_response(response)
  File "/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/wpull/recorder.py", line 412, in pre_response
    print(response.status_code, response.status_reason)
UnicodeEncodeError: 'ascii' codec can't encode character '\xfe' in position 2: ordinal not in range(128)
INFO FINISHED.
INFO Time length: 0.1 seconds.
INFO Downloaded: 0 files, 0 bytes.
INFO Exiting with status 2.
200 
```

session.log:

```
2014-01-25 20:25:42,121: Start session
2014-01-25 20:25:42,122: Create environment variable PYTHONPATH: (len=106)
2014-01-25 20:25:42,122: Environment: {'PYTHONPATH': '/home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/test/fuzz_fusil/../..'}
2014-01-25 20:25:42,123: Stdin: /dev/null
2014-01-25 20:25:42,123: Stdout filename: /home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/test/fuzz_fusil/fusil/session-4/stdout
2014-01-25 20:25:42,123: Create process: ['/usr/bin/python3', '-m', 'wpull', '127.0.0.1:8898', '--timeout', '2.0', '--tries', '1']
2014-01-25 20:25:42,123: Working directory: /home/chris/data-chris-aspire-5741/Documents/programming/wget-remake/wget-remake.git/test/fuzz_fusil/fusil/session-4
2014-01-25 20:25:42,131: Process identifier: 15504
2014-01-25 20:25:42,416: Accept client
2014-01-25 20:25:42,416: New client: <ServerClient (host 127.0.0.1, port 37954)>
2014-01-25 20:25:42,419: Read data from <ServerClient (host 127.0.0.1, port 37954)>
2014-01-25 20:25:42,431: request choice: 1
2014-01-25 20:25:42,431: Mangle content: YES
2014-01-25 20:25:42,432: Mangled data: bytearray(b"\'* the uncoding o\xff\x7fthis\x86d/cuoent s\x80\x00\x00\x00d be\x95in\xe2Shift_JIS\xff\xff\xff\xfe/\x80\xb7\x8e\x84\x82\xcd\x83K\x83\xa9\x83X\x82\xf0\x90\xf9\x82\xd7\x82\xe7\x82\xea\x00\xfef\xb7\x81B\xd4\xbb\x9e\xea\xbf\xcd\x8e\x84\xff\x80\x8f\x9d\x82\xfe\xff\xff\xff\xff\x00\xb9\x00\x00\x00\x80 */\nbod\x08\xf6;\x7f    backgrfu\xfe\xff\xff\xffma\x80e\xff\xfeurl(g/\x98\x00\x8e\x9a\x89\xbb\x82\xaf.Png\'\xff\xfe\n}\n\n")
2014-01-25 20:25:42,432: Mangle header: YES
2014-01-25 20:25:42,432: Mangled data: bytearray(b'HTTP/1.0 200 O\x00\xfe\nServer: Fusil\xff\xff\xff\xfeagma: no-cache\r\nContent-Type: text/html\r\nContent-Length: 170\r\n\r\n')
2014-01-25 20:25:42,432: Close socket
2014-01-25 20:25:42,432: Client closed: <ServerClient (host 127.0.0.1, port 37954)>
2014-01-25 20:25:42,440: Match pattern 'exception' (score 100.0%) in 'Requesting http://127.0.0.1:8898/robots.txt... ERROR Fatal exception.'
2014-01-25 20:25:42,442: - <WatchStdout 'watch:stdout'> score: 100.0%
2014-01-25 20:25:42,446: End of session: score=100.0%, duration=0.326 second
```

TL;DR: ProgressRecorderSession.pre_response should be printing to self._stream not stdout!
